### PR TITLE
Corrected script for some item

### PR DESCRIPTION
**Pull request by idk-whoami**
* **Addressed Issue(s)**: 

* **Server Mode**: re

* **Description of Pull Request**:

### CODE REVIEW

File: db/re/item_combo_db.txt [hunk 1]
```diff
@@ -2742,3 +2741,16 @@
...
 24744:24584:24585:24586:24587:24588,{ bonus2 bMagicAtkEle,Ele_All,15; bonus2 bIgnoreDefRaceRate,RC_All,70; bonus2 bIgnoreMdefRaceRate,RC_All,70; bonus2 bIgnoreDefRaceRate,RC_Player_Human,-70; bonus2 bIgnoreMdefRaceRate,RC_Player_Human,-70; bonus2 bIgnoreDefRaceRate,RC_Player_Doram,-70; bonus2 bIgnoreMdefRaceRate,RC_Player_Doram,-70; bonus bShortAtkRate,15; bonus bDelayrate,-15; bonus bVariableCastrate,-10; bonus2 bSkillAtk,"RK_IGNITIONBREAK",15; bonus2 bSkillAtk,"SO_PSYCHIC_WAVE",15; }
+2114:15001:5122,{ bonus bStr,2; bonus bDef,5; bonus bMdef,5; if(BaseClass==Job_Swordman) bonus bDef,6; }
+2114:15001:5124,{ bonus bDef,2-getequiprefinerycnt(EQI_HAND_L)-getequiprefinerycnt(EQI_HEAD_TOP); bonus bMdef,5+getequiprefinerycnt(EQI_HAND_L)+getequiprefinerycnt(EQI_HEAD_TOP); }
+2455:2559:5544:15007,{ bonus bBaseAtk,20; bonus bMatk,20; bonus bAspdRate,10; }
```
Comment (Atemo) [inline db/re/item_combo_db.txt:2746]: I can't find the description giving theses bonus. Link?

File: db/re/item_db_equip.yml [hunk 2]
```diff
@@ -22891,7 +22881,12 @@ Body:
     Script: |
       bonus bMdef,5;
       .@r = getrefine();
-      bonus3 bAutoSpellWhenHit,"NPC_DRAGONFEAR",1,(.@r<=6)?(10):((.@r<=8)?(20):(30));
+      if (.@r>=7)
+         bonus3 bAutoSpellWhenHit,"NPC_DRAGONFEAR",2,20;
+      else if (.@r>=9)
+         bonus3 bAutoSpellWhenHit,"NPC_DRAGONFEAR",3,30;
```
Comment (Atemo) [inline db/re/item_db_equip.yml:22887]:
```suggestion
      if (.@r>=9)
         bonus3 bAutoSpellWhenHit,"NPC_DRAGONFEAR",3,30;
      else if (.@r>=7)
         bonus3 bAutoSpellWhenHit,"NPC_DRAGONFEAR",2,20;
```

File: db/re/item_db_equip.yml [hunk 3]
```diff
@@ -23265,6 +23252,40 @@ Body:
...
+      Left_Hand: true
+    ArmorLevel: 1
+    EquipLevelMin: 60
+    Refineable: true
+    Script: |
+      .@r = getrefine();
+      if (.@r >=7)
+         autobonus "{ bonus bDelayrate,-10-(5*(.@r-6)); }",50,5000,BF_MAGIC;
+      else
+         autobonus "{ bonus bDelayrate,-10; }",50,5000,BF_MAGIC;
```
Comment (Atemo) [inline db/re/item_db_equip.yml:23288]: "When hit by" it should be autobonus2
The `.@r` inside your autobonus is not defined, the bonus inside the string is running independently of the item script
```suggestion
      if (.@r >=7)
         autobonus2 "{ .@r = getrefine(); bonus bDelayrate,-10-(5*(.@r-6)); }",50,5000,BF_MAGIC;
      else
         autobonus2 "{ bonus bDelayrate,-10; }",50,5000,BF_MAGIC;
```

File: db/re/item_db_equip.yml [hunk 4]
```diff
@@ -23314,6 +23335,24 @@ Body:
...
+    Locations:
+      Left_Hand: true
+    ArmorLevel: 1
+    EquipLevelMin: 90
+    Refineable: true
+    Script: |
+      bonus bUnbreakableShield;
+      bonus bAspdRate,-3;
+      if (getrefine()>=8)
+         bonus bNoKnockback;
```
Comment (Atemo) [inline db/re/item_db_equip.yml:23355]: For me the malus bAspdRate should be when refine is >= 8
```suggestion
      if (getrefine()>=8) {
         bonus bAspdRate,-3;
         bonus bNoKnockback;
      }
```

File: db/re/item_db_equip.yml [hunk 5]
```diff
@@ -138193,6 +138287,77 @@ Body:
...
+      bonus2 bSkillAtk,"NC_AXETORNADO",30*.@lvl;
+      bonus2 bSkillAtk,"NC_AXEBOOMERANG",20*.@lvl;
+      bonus bIgnoreDefRace,20*getskilllv("NC_MAGMA_ERUPTION");
+      if (getskilllv("NC_MAGICDECOY") == 5) {
+         bonus bAspdRate,10;
+         bonus bMaxHP,150*BaseLevel;
+      }
+      if (.@r>=5) {
+         bonus bMaxHPrate,3;
+         bonus bMaxSPrate,3;
```
Comment (Atemo) [inline db/re/item_db_equip.yml:138317]:
```suggestion
         bonus bMaxHPrate,7;
         bonus bMaxSPrate,7;
```

File: db/re/item_db_equip.yml [hunk 6]
```diff
@@ -138193,6 +138287,77 @@ Body:
...
```
Comment (Atemo) [inline db/re/item_db_equip.yml:138353]:
```suggestion
         bonus bMaxHPrate,7;
         bonus bMaxSPrate,7;
```

**Review by Atemo — CHANGES_REQUESTED** [review]

File: db/re/item_db_equip.yml (hunk 6 above)
Comment (idk-whoami) [reply db/re/item_db_equip.yml:138353]: Thanks !

File: db/re/item_db_equip.yml (hunk 5 above)
Comment (idk-whoami) [reply db/re/item_db_equip.yml:138317]: Thanks !

File: db/re/item_db_equip.yml (hunk 4 above)
Comment (idk-whoami) [reply db/re/item_db_equip.yml:23355]: Thanks !

File: db/re/item_db_equip.yml (hunk 3 above)
Comment (idk-whoami) [reply db/re/item_db_equip.yml:23288]: Oh my apologies and thanks for the knowledge.

File: db/re/item_db_equip.yml (hunk 2 above)
Comment (idk-whoami) [reply db/re/item_db_equip.yml:22887]: Thanks !

File: db/re/item_combo_db.txt (hunk 1 above)
Comment (idk-whoami) [reply db/re/item_combo_db.txt:2746]: Hidden bonus.  Reference from thRO

image

https://youtu.be/05PDJOlXT90

https://youtu.be/PpcWQhTXi4g

**Comment by Atemo** [comment]
Good work, thanks!

**Pull request merged**